src/main/java/teammates/common/util/SanitizationHelper.java

```diff
@@ -314,36 +336,30 @@ private static String trimIfNotNull(String string) {
      * @param text
      * @return safer version of the text for XPath
      */
-    public static String convertStringForXPath(String text) {
+    public static String sanitizeStringForXPath(String text) {
         StringBuilder result = new StringBuilder();
-        int startPos = 0;
-        int i = 0;
-        while (i < text.length()) {
-            while (i < text.length() && text.charAt(i) != '\'') {
-                i++;
-            }
-            if (startPos < i) {
-                result.append('\'').append(text.substring(startPos, i)).append("',");
-                startPos = i;
+        int startOfChain = 0;
+        int textLength = text.length();
+        boolean isSingleQuotationChain = false;
+        // currentPos iterates one position beyond text length to include last chain
+        for (int currentPos = 0; currentPos <= textLength; currentPos++) {
+            boolean isChainBroken = currentPos >= textLength
+                    || isSingleQuotationChain && text.charAt(currentPos) != '\''
+                    || !isSingleQuotationChain && text.charAt(currentPos) == '\'';
```

Comment:
Here as well, I suggest the `||` be in line with the `c` of `currentPos` for readability